app/js/sumo_db.js

```diff
@@ -154,6 +153,21 @@
       endpoint += '?format=json'; // TODO bug 1088014
 
       return request(endpoint, 'GET').then(JSON.parse);
+    },
+
+    update_preference: function(user, setting) {
+      var endpoint = API_V2_BASE + 'user/';
+      endpoint += user.username + '/';
+
+      var delete_setting = endpoint + 'delete_setting/';
+      delete_setting += '?format=json'; // TODO bug 1088014
+      return request_with_auth(delete_setting, 'POST', setting)
+        .then(function(response) {
```

Comment:
In general, I wouldn't mind landing a workaround until the server side parts is fixed. But this workaround fails because we get a 404 from the delete call when we've never created those settings. Let's wait for bug 1118037 to land.